### PR TITLE
bf: ZENKO-763 fix object tag replication

### PR DESCRIPTION
**Pull request by jonathan-gramain**
Add extra logic to consume mongo oplog correctly (commit message contains more details)

### CODE REVIEW

**Comment by jonathan-gramain** [comment]
I will rework a bit as some other log entry cases are unclear, and add some unit tests.

**Comment by jonathan-gramain** [comment]
@bert-e wait

**Review by bennettbuchanan — APPROVED** [review]
Thanks for this @jonathan-gramain! :shipit:

File: lib/storage/metadata/mongoclient/LogConsumer.js [hunk 1]
```diff
@@ -63,16 +57,39 @@ class ListRecordStream extends stream.Transform {
         }
 
         const dbName = itemObj.ns.split('.');
+        let entry;
+        if (itemObj.op === 'i') {
+            entry = {
+                type: 'put',
+                key: itemObj.o._id,
```
Comment (rahulreddy) [inline lib/storage/metadata/mongoclient/LogConsumer.js]: We should check for the existence of `o` and `o2` properties before accessing their children. I know that the format is probably guaranteed by Mongodb but it's best if we check for the properties cleanly.

**Review by JianqinWang — APPROVED** [review]
thank you  @jonathan-gramain !

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: lib/storage/metadata/mongoclient/LogConsumer.js (hunk 1 above)
Comment (jonathan-gramain) [reply lib/storage/metadata/mongoclient/LogConsumer.js]: done

File: lib/storage/metadata/mongoclient/LogConsumer.js [hunk 2]
```diff
@@ -151,15 +66,28 @@ class LogConsumer {
...
+                // previous 'info' event
+                startSeq = JSON.parse(params.startSeq);
+            } catch (err) {
+                this.logger.error('malformed startSeq', {
+                    startSeq: params.startSeq,
+                });
+                // start over
+                startSeq = { ts: 0 };
+            }
+        } else {
```
Comment (bennettbuchanan) [inline lib/storage/metadata/mongoclient/LogConsumer.js]: Minor: we can avoid the else block here if we initially set `startSeq` to `{ ts: 0 }`.

**Review by bennettbuchanan — APPROVED** [review]

File: lib/storage/metadata/mongoclient/LogConsumer.js (hunk 2 above)
Comment (jonathan-gramain) [reply lib/storage/metadata/mongoclient/LogConsumer.js]: done

**Review by JianqinWang — APPROVED** [review]

**Comment by jonathan-gramain** [comment]
E2E tests have some unrelated flakiness issues during install phase, but all replication tests pass (including the re-enabled CRR tagging tests, see https://github.com/scality/Zenko/pull/227). In this respect it should be good to merge, and updated E2E tests will run again in zenko with updated images.

One build where CRR tests pass: https://eve.devsca.com/github/scality/zenko/#/builders/10/builds/3532

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**